### PR TITLE
Update dem stitcher api

### LINKED ISSUE #22: Update Dem Stitcher API in Dockerized TopsApp and Pull from Github
- install directly from pip (avoid temporarily the pypi release)
- ensure the new API is correct

### PR DESCRIPTION
**Pull request by cmarshak**
Fixes #22.

@gracebato - I tried this branch using the problematic pair. I believe it fixes.

Important notes:

- uses 0 for nodata areas
- The `gdal` metadata `nodata` is set to None.

```
isce2_topsapp --reference-scenes S1B_IW_SLC__1SDV_20211004T161549_20211004T161616_028988_037594_4DF8 --secondary-scenes S1A_IW_SLC__1SDV_20210928T161630_20210928T161657_039884_04B811_264A
```

Gets browse imagery

image

### CODE REVIEW

**Comment by cmarshak** [comment]
Hit a snag on build.

we are stuck on environment install issues related to this line: https://github.com/ACCESS-Cloud-Based-InSAR/DockerizedTopsApp/blob/update_dem_stitcher_api/environment.yml#L25

Gives us this error:

```....
     raise ValueError('Newlines are not allowed')
  ValueError: Newlines are not allowed
  ----------------------------------------
WARNING: Discarding git+https://github.com/ACCESS-Cloud-Based-InSAR/dem_stitcher.git. Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.

failed

CondaEnvException: Pip failed```

**Comment by cmarshak** [comment]
@gracebato - the fix that Joe found definitely works. I am going to wait until you review and then we can merge.

I am going to also specify python version >= 3.8 now.

**Review by gracebato — APPROVED** [review]
Looks good to me. I tested the docker build/setup and it is now working both in the cluster and in my local machine (macOS Catalina).

**Pull request merged**
